cmd_rank.lua

```diff
@@ -0,0 +1,59 @@
+function HandleRankCommand(Split, Player)
+	local Response
+	local PlayerName = Split[2]
+	local NewRank = Split[3]
+	
+	local InformLoadRank = function(OtherPlayer)
+		if PlayerName == OtherPlayer:GetName() then
+			local AuthorName = "the server console"
+			
+			if Player then
+				AuthorName = "\"" .. Player:GetName() .. "\""
+			end
+
+			OtherPlayer:SendMessageInfo("You were assigned the rank " .. NewRank .. " by " .. AuthorName)
+			OtherPlayer:LoadRank()
+		end
+	end
+
+	if not PlayerName then
+		Response = SendMessage(Player, "Usage: " .. Split[1] .. " <player> [rank]")
+	else
+		-- Translate the PlayerName to a UUID:
+		local PlayerUUID = GetPlayerUUID(PlayerName)
+
+		if not PlayerUUID or string.len(PlayerUUID) ~= 32 then
+			return true, SendMessage(Player, "There is no player with the name \"" .. PlayerName .. "\"")
```

Comment:
Why break the pattern here, I don't think the benefit from making this a guard clause overrides the confusion.

Reply:
Logic flies out the window sometimes when you stare at code for too long...